packages/cardpay-sdk/sdk/utils/merchant.ts

```diff
@@ -33,11 +33,10 @@ export const generateMerchantPaymentUrl = ({
   network,
   currency,
 }: MerchantPaymentURLParams) => {
-  const handleAmount = amount ? `amount=${amount}&` : '';
-  const handleCurrency = currency ? `${handleAmount ? '&' : ''}currency=${currency}` : '';
+  const handleAmountAndCurrency = currency ? `${amount ? `amount=${amount}&` : ''}currency=${currency}` : '';
   const https = isUniversalDomain(domain) ? 'https://' : '';
 
-  return `${https}${domain}/pay/${network}/${merchantSafeID}?${handleAmount}${handleCurrency}`;
+  return `${https}${domain}/pay/${network}/${merchantSafeID}?${handleAmountAndCurrency}`;
```

Comment:
This '?' probably isn't necessary if we don't append params, but I think this is a minor detail.